### PR TITLE
chore: add viz-echo example

### PR DESCRIPTION
**Pull request by fundon**
Viz supports a [`ServiceHandler`](https://docs.rs/viz/latest/viz/handler/struct.ServiceHandler.html) wrapper to handle socketioxide service in v0.7.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 target
-.env
\ No newline at end of file
+Cargo.lock
```
Comment (Totodore) [inline .gitignore:2]: I understand that it should be better to ignore the lock file and I agree to do it. But it should be done on another PR, it is not the goal of this one.

File: examples/viz-echo/viz_echo.rs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    svc: S,
+}
+
+#[async_trait]
+impl<S, I, O> Handler<Request<I>> for SocketIoHandler<S>
+where
+    I: Body + Send + Unpin + std::fmt::Debug + 'static,
+    I::Error: std::fmt::Debug,
+    I::Data: Send,
+    // O: Body + Send + 'static,
```
Comment (Totodore) [inline examples/viz-echo/viz_echo.rs:30]: Remove commented lines

File: examples/viz-echo/viz_echo.rs [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            .map(|resp| {
+                resp.map(|body| {
+                    body.map_frame(|f| f.map_data(Into::into))
+                        .map_err(Into::into)
+                        // .boxed_unsync()
+                        .boxed()
+                })
+                .map(Into::into)
+            })
+            .map_err(Into::into)
```
Comment (Totodore) [inline examples/viz-echo/viz_echo.rs:53]: This could be flattened

File: socketioxide/src/io.rs [hunk 4]
```diff
@@ -778,13 +778,14 @@ mod tests {
...
+        // assert!(io.get_default_op().is_broadcast());
+        // assert!(io.to("room1").is_broadcast());
+        // assert!(io.within("room1").is_broadcast());
+        // assert!(io.except("room1").is_broadcast());
+        // assert!(io.local().is_broadcast());
+        // assert!(io.timeout(Duration::from_secs(5)).is_broadcast());
+        // assert!(io.bin(vec![vec![1, 2, 3, 4]]).is_broadcast());
+        // assert!(io.of("/").unwrap().is_broadcast());
     }
 }
```
Comment (Totodore) [inline socketioxide/src/io.rs:791]: This deserve some explanations

**Review by Totodore — CHANGES_REQUESTED** [review]
You are modifying many things that we don't need to or that are not related to the subject of the PR. 
If it is the case, add explanations or make other prs please :)

Can you also add an explanations of what you changed in your pr description.

**Comment by fundon** [comment]
Thank you. I split some small PRs.

File: .gitignore (hunk 1 above)
Comment (fundon) [reply .gitignore:2]: Related to #204.

**Review by Totodore — APPROVED** [review]
